redfish-core/lib/cable.hpp

```diff
@@ -1,9 +1,208 @@
 #pragma once
 #include <boost/container/flat_map.hpp>
+#include <utils/chassis_utils.hpp>
 #include <utils/json_utils.hpp>
 
 namespace redfish
 {
+
+inline void linkAssociatedCableWithPort(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const std::string& cableObjectPath, const std::string& association,
+    const std::string& jsonKey)
+{
+    crow::connections::systemBus->async_method_call(
+        [asyncResp, cableObjectPath,
+         jsonKey](const boost::system::error_code ec,
+                  const std::variant<std::vector<std::string>>& endpoints) {
+            if (ec)
+            {
+                if (ec.value() == EBADR)
+                {
+                    // This cable have no association.
+                    BMCWEB_LOG_DEBUG << "No association found";
+                    return;
+                }
+                BMCWEB_LOG_ERROR << "DBUS response error" << ec.message();
+                messages::internalError(asyncResp->res);
+                return;
+            }
+
+            const std::vector<std::string>* value =
+                std::get_if<std::vector<std::string>>(&(endpoints));
+
+            if (value == nullptr)
+            {
+                BMCWEB_LOG_DEBUG << "Error getting cable association!";
+                messages::internalError(asyncResp->res);
+                return;
+            }
+
+            if ((*value).size() < 1)
+            {
+                BMCWEB_LOG_DEBUG << "No association found for cable";
+                messages::internalError(asyncResp->res);
+                return;
+            }
+
+            nlohmann::json& linkArray =
+                asyncResp->res.jsonValue["Links"][jsonKey];
+            linkArray = nlohmann::json::array();
+
+            for (const auto& fullPath : *value)
+            {
+                sdbusplus::message::object_path path(fullPath);
+                std::string leaf = path.filename();
+                if (leaf.empty())
+                {
+                    continue;
+                }
+                std::string endpointLeaf = path.parent_path().filename();
+                if (endpointLeaf.empty())
+                {
+                    continue;
+                }
+
+                //  insert/create link suing endpointLeaf
+                endpointLeaf += "/Ports/";
+                endpointLeaf += leaf;
+                linkArray.push_back(
+                    {{"@odata.id",
+                      "/redfish/v1/Systems/system/FabricAdapters/" +
+                          endpointLeaf}});
+            }
+        },
+        "xyz.openbmc_project.ObjectMapper", cableObjectPath + association,
```

Comment:
Does cableObjectPath has ending '/' OR association string starts with a '/'. I'm not sure, just a clarification